R/build-logo.R

```diff
@@ -7,19 +7,57 @@ build_logo <- function(pkg = ".") {
 
   file_copy_to(pkg, logo_path, from_dir = path_dir(logo_path))
 
-  if (!requireNamespace("magick", quietly = TRUE)) {
-    message("magick not available, using default pkgdown favicon.ico")
+  cat_line("Creating ", dst_path("favicon.ico"))
 
-    if (!file_exists(path(pkg$dst_path, "favicon.ico"))) {
-      file_copy(path_pkgdown("assets", "favicon.ico"), pkg$dst_path)
-    }
-    return()
+  logo <- readBin(logo_path, what = "raw", n = fs::file_info(logo_path)$size)
+
+  logo_base64 <- openssl::base64_encode(logo)
+
+  json_request <- list(
+    "favicon_generation" = list(
+      "api_key" = "87d5cd739b05c00416c4a19cd14a8bb5632ea563",
+      "master_picture" = list(
+        "type"= "inline",
+        "content"= logo_base64
+      ),
+      "favicon_design" = list(
+        "desktop_browser" = list(),
+        "ios" = list(
+          "picture_aspect" = "no_change",
+          "assets" = list(
+            "ios6_and_prior_icons" = FALSE,
+            "ios7_and_later_icons" = TRUE,
+            "precomposed_icons" = FALSE,
+            "declare_only_default_icon" = TRUE
+          )
+        )
+      )
+    )
+  )
+
+  # It may take some time to generate the whole favicon set so we need to set
+  # a high timeout value.
+  request <- httr::POST("https://realfavicongenerator.net/api/favicon",
```

Comment:
Oh wow, if it needs to be that long, this definitely should be something the user needs to explicitly call.

Reply:
Can you please indent like http://style.tidyverse.org/syntax.html#long-lines?